app/bundles/LeadBundle/Helper/IdentifyCompanyHelper.php

```diff
@@ -62,9 +64,12 @@ public static function identifyLeadsCompany($parameters, Lead $lead, CompanyMode
 
             if (!empty($company)) {
                 //check if lead is already assigned to company
-                $companyLeadRepo = $companyModel->getCompanyLeadRepository();
-                if (empty($companyLeadRepo->getCompaniesByLeadId($lead->getId(), $company['id']))) {
-                    $leadAdded = true;
+                $companyEntity = $companyModel->getEntity($company['id']);
```

Comment:
This results in two queries to get the same information: line 61/63 for a DBAL query `$companyRepo->identifyCompany()` then a second query to hydrate an entity.  Can we fetch an entity to start with then hydrate an array for $company with values from the entity rather than querying twice?